sdk/core/Azure.Core/src/Shared/ConnectionString.cs

```diff
@@ -0,0 +1,248 @@
+        {
+            var segmentStart = -1;
+            var segmentEnd = 0;
+
+            while (TryGetNextSegment(connectionString, segmentSeparator, ref segmentStart, ref segmentEnd))
+            {
+                if (segmentStart == segmentEnd)
+                {
+                    if (segmentStart == 0)
+                    {
+                        throw new InvalidOperationException($"Connection string starts with separator '{segmentSeparator}'.");
+                    }
+
+                    throw new InvalidOperationException($"Connection string contains two following separators '{segmentSeparator}'.");
+                }
+
+                var kvSeparatorIndex = connectionString.IndexOf(keyValueSeparator, segmentStart, segmentEnd - segmentStart, StringComparison.Ordinal);
+                if (kvSeparatorIndex != -1)
+                {
+                    ValidateKeyValue(connectionString, segmentStart, segmentEnd, allowEmptyValues, allowWhitespaces, kvSeparatorIndex);
+                }
+                else
+                {
+                    if (allowFlags)
+                    {
+                        ValidateWhitespaces(connectionString, segmentStart, segmentEnd, allowWhitespaces, "key");
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Connection string segment '{connectionString.Substring(segmentStart, segmentEnd - segmentStart)}' should have a value.");
+                    }
+                }
+            }
+        }
+
+        private static void ValidateKeyValue(string connectionString, int segmentStart, int segmentEnd, bool allowEmptyValues, bool allowWhitespaces, int kvSeparatorIndex)
+        {
+            if (segmentStart == kvSeparatorIndex)
+            {
+                throw new InvalidOperationException($"Segment '{connectionString.Substring(segmentStart, kvSeparatorIndex - segmentStart)}' has no key.");
+            }
+
+            if (!allowEmptyValues && kvSeparatorIndex + 1 == segmentEnd)
+            {
+                throw new InvalidOperationException($"Key '{connectionString.Substring(segmentStart, kvSeparatorIndex - segmentStart)}' has no value.");
+            }
+
+            ValidateWhitespaces(connectionString, segmentStart, kvSeparatorIndex, allowWhitespaces, "key");
+            ValidateWhitespaces(connectionString, kvSeparatorIndex + 1, segmentEnd, allowWhitespaces, "value");
+        }
+
+        private static void ValidateWhitespaces(string connectionString, int start, int end, bool allowWhitespaces, string name)
+        {
+            if (allowWhitespaces)
+            {
+                return;
+            }
+
+            if (char.IsWhiteSpace(connectionString[start]))
+            {
+                throw new InvalidOperationException($"Start of the {name} '{connectionString.Substring(start, end - start)}' shouldn't contain spaces.");
```

Comment:
Do we really want to enforce this? When would trimming the name for customers be wrong?

Reply:
I've seen such requirements couple times in practice, but since it's an internal type, I can remove the redundant check. Up to you.

Reply:
I wouldn't put limitations on inputs if they are not essential for the functionality.

Reply:
Done.